mne/utils.py

```diff
@@ -252,6 +252,42 @@ def sum_squared(X):
     return np.dot(X_flat, X_flat)
 
 
+def warn(message, category=RuntimeWarning):
+    """Emit a warning with trace outside the mne namespace
+
+    This function takes arguments like warnings.warn, and sends messages
+    using both warnings.warn and logger.warn. It also uses a default type
+    of RuntimeWarning.
+
+    Parameters
+    ----------
+    message : str
+        Warning message.
+    category : instance of Warning
+        The warning class.
+    """
+    import mne
+    root_dir = op.dirname(mne.__file__)
+    stacklevel = 1
+    frame = None
+    stack = inspect.stack()
+    last_file = ''
+    for fi, frame in enumerate(stack):
+        frame = frame[1]
```

Comment:
nitpick: maybe rename this to `filename = frame[1]`? You are selecting part of the `frame` array here.